crates/matrix-sdk-ui/tests/integration/timeline/thread.rs

```diff
@@ -0,0 +1,308 @@
+
+    let room_id = room_id!("!a:b.c");
+    let sender_id = user_id!("@alice:b.c");
+
+    let factory = EventFactory::new().room(room_id).sender(sender_id);
+
+    let thread_root_event_id = owned_event_id!("$root");
+
+    server
+        .mock_room_event()
+        .match_event_id()
+        .ok(factory
+            .text_msg("Thread root")
+            .sender(sender_id)
+            .event_id(&thread_root_event_id)
+            .into())
+        .mock_once()
+        .mount()
+        .await;
+
+    let batch1 = vec![
+        factory.text_msg("Threaded event 4").event_id(event_id!("$3")).into_raw_sync().cast(),
+        factory.text_msg("Threaded event 3").event_id(event_id!("$4")).into_raw_sync().cast(),
+    ];
+    let batch2 = vec![
+        factory.text_msg("Threaded event 2").event_id(event_id!("$1")).into_raw_sync().cast(),
+        factory.text_msg("Threaded event 1").event_id(event_id!("$2")).into_raw_sync().cast(),
+    ];
+
+    server
+        .mock_room_relations()
+        .match_target_event(thread_root_event_id.clone())
+        .ok(RoomRelationsResponseTemplate::default().events(batch1).next_batch("next_batch"))
+        .mock_once()
+        .mount()
+        .await;
+
+    server
+        .mock_room_relations()
+        .match_target_event(thread_root_event_id.clone())
+        .match_from("next_batch")
+        .ok(RoomRelationsResponseTemplate::default().events(batch2))
+        .mock_once()
+        .mount()
+        .await;
+
+    let room = server.sync_joined_room(&client, room_id).await;
+
+    let timeline = Timeline::builder(&room)
+        .with_focus(TimelineFocus::Thread { root_event_id: thread_root_event_id, num_events: 1 })
+        .build()
+        .await
+        .unwrap();
+
+    let (items, mut timeline_stream) = timeline.subscribe().await;
+
+    assert_eq!(
+        items[1].as_event().unwrap().content().as_message().unwrap().body(),
+        "Threaded event 3"
+    );
+    assert_eq!(
+        items[2].as_event().unwrap().content().as_message().unwrap().body(),
+        "Threaded event 4"
+    );
+    assert_pending!(timeline_stream);
+
+    let hit_start = timeline.paginate_backwards(100).await.unwrap();
+    assert!(hit_start);
+
+    timeline_stream.next().await;
+
+    let items = timeline.items().await;
+
+    assert_eq!(items[1].as_event().unwrap().content().as_message().unwrap().body(), "Thread root");
+
+    assert_eq!(
+        items[2].as_event().unwrap().content().as_message().unwrap().body(),
+        "Threaded event 1"
+    );
+
+    assert_eq!(
+        items[3].as_event().unwrap().content().as_message().unwrap().body(),
+        "Threaded event 2"
+    );
+    assert_eq!(
+        items[4].as_event().unwrap().content().as_message().unwrap().body(),
+        "Threaded event 3"
+    );
+    assert_eq!(
+        items[5].as_event().unwrap().content().as_message().unwrap().body(),
+        "Threaded event 4"
+    );
+}
+
+#[async_test]
+async fn test_thread_live_updates() {
```

Comment:
This test doesn't do much yet, right? Maybe we can get rid of it until it does :)

Reply:
Fair, I didn't know whether we want one that will fail later or the opposite tbh.